src/shogun/base/SGObject.cpp

```diff
@@ -471,6 +478,68 @@ void CSGObject::save_serializable_post() throw (ShogunException)
 	m_save_post_called = true;
 }
 
+void CSGObject::saveBinary(char* filename) const
+{
+	std::ofstream os(filename);
+	cereal::BinaryOutputArchive archive(os);
+	archive(*this);
+}
+
+void CSGObject::saveJSON(char* filename) const
```

Comment:
Not sure, I would find a single method that takes an enum cleaner here. But that is my taste
@vigsterkr any comments?